hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -547,4 +541,42 @@ public void setConf(Configuration conf) {
   public Configuration getConf() {
     return conf;
   }
+
+  @VisibleForTesting
+  HiveClientPool clientPool() {
+    synchronized (CLIENT_POOL_CACHE) {
+      String metastoreUri = conf.get(HiveConf.ConfVars.METASTOREURIS.varname, "");
+      Pair<HiveClientPool, Long> cacheEntry = CLIENT_POOL_CACHE.getIfPresent(metastoreUri);
+      HiveClientPool clientPool = cacheEntry == null ? new HiveClientPool(clientPoolSize, conf) : cacheEntry.first();
+      CLIENT_POOL_CACHE.put(metastoreUri, Pair.of(clientPool, System.currentTimeMillis() + evictionInterval));
+      return clientPool;
+    }
+  }
+
+  private void scheduleCacheCleaner() {
```

Comment:
I think it would be cleaner to always keep the pool around and never close it until the cache is cleaned up. Then the pool would always be shared and could be kept as an instance variable inside the catalog. It would also make cleanup less coarse. We could expire individual connections that are unused instead of expiring whole pools. That aligns better with usage patterns and keeps the complexity inside the pool rather than in the cache.

Reply:
@rdblue I made a few changes compared to the original version of the HiveClientPoolCache. Now the cleanup and update are completely delegated to the cache itself. Could you please review it? Thanks